frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1090,6 +1112,73 @@ function ReaderBookmark:setBookmarkNote(item, from_highlight, is_new_note, new_t
     self.input:onShowKeyboard()
 end
 
+function ReaderBookmark:editHighlightedText(item)
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = _("Edit highlighted text"),
+        description = "   " .. self._getDialogHeader(item),
+        input = item.notes,
+        allow_newline = true,
+        add_scroll_buttons = true,
+        use_available_height = true,
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(input_dialog)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    is_enter_default = true,
+                    callback = function()
+                        self:setHighlightedText(item, input_dialog:getInputText())
+                        UIManager:close(input_dialog)
+                    end,
+                },
+            }
+        },
+    }
+    UIManager:show(input_dialog)
+    input_dialog:onShowKeyboard()
+end
+
+function ReaderBookmark:setHighlightedText(item, text)
+    local edited
+    if text then
+        edited = true
+    else -- reset to selected text
+        if self.ui.rolling then
+            text = self.ui.document:getTextFromXPointers(item.pos0, item.pos1)
+        else
+            text = self.ui.document:getTextFromPositions(item.pos0, item.pos1).text
+        end
+    end
+    -- highlight
+    local hl = self.ui.highlight:getHighlightByDatetime(item.datetime)
+    hl.text = text
+    hl.edited = edited
```

Comment:
May be `text_edited` ? (An "edited" bookmark is a bit vague and could mean other things.)